packages/providers/src/transaction-request/transaction-request.ts

```diff
@@ -398,10 +400,11 @@ export class ScriptTransactionRequest extends BaseTransactionRequest {
   /** determined bytes offset for start of script data */
   bytesOffset: number | undefined;
 
-  constructor({ script, scriptData, ...rest }: ScriptTransactionRequestLike = {}) {
+  constructor({ script, scriptData, bytesOffset, ...rest }: ScriptTransactionRequestLike = {}) {
     super(rest);
     this.script = arrayify(script ?? returnZeroScript.bytes);
     this.scriptData = arrayify(scriptData ?? returnZeroScript.encodeScriptData());
+    this.bytesOffset = bytesOffset;
```

Comment:
I miss unit tests for this change:
 1. What's the utility of the `bytesOffset` variable?
 1. What changed after you started setting it on the constructor?

Reply:
```
/** determined bytes offset for start of script data */
bytesOffset: number | undefined;
```

`bytesOffset` is the offset to find script data inside transaction raw payload.

before this change, when creating a new transaction (using `transactionRequestify` for instance), it was not getting the bytesOffset, resulting on
1- a bug(fixed using the `?` change in this PR).
2- incorrect encoding when calling contracts (you can see its usage [here](https://github.com/FuelLabs/fuels-ts/pull/891/files#diff-3409481ab69cbec51784c01628082462a0c87c332fd2e0cdf97fb93d9a59b2fdR25)

only change 1 would stop breaking the code, but encoding would continue incorrect

now `bytesOffset` is saved properly in transactionRequest

Reply:
> 2- incorrect encoding when calling contracts

@LuizAsFight will this fix #881 by any chance? Or is this unrelated?

Reply:
@LuizAsFight It seems the only places using `bytesOffset` are the `constructor` and the `setScript` method.

From this, I'd presume this variable currently has no function.

What am I missing?

https://github.com/FuelLabs/fuels-ts/blob/c8736e8c24fca01b497e6d44fbeb087cd6c570d8/packages/providers/src/transaction-request/transaction-request.ts#L403-L408

https://github.com/FuelLabs/fuels-ts/blob/c8736e8c24fca01b497e6d44fbeb087cd6c570d8/packages/providers/src/transaction-request/transaction-request.ts#L442-L449

Reply:
Oh, never mind, VSCode wasn't finding all references properly; it is used [here](https://github.com/FuelLabs/fuels-ts/blob/59aabffe76e7f03f1a4082b09f302a293a322a42/packages/program/src/functions/base-invocation-scope.ts#L28) and [here](https://github.com/FuelLabs/fuels-ts/blob/c7cb8ac2e268b860a41d29927814c24339f8514a/packages/program/src/functions/invocation-scope.ts#L36).

Looking forward to the answer to @Dhaiwat10's [question](https://github.com/FuelLabs/fuels-ts/pull/890#discussion_r1163462497).

Reply:
hey @Dhaiwat10 , no, it's not related